apps/E2E/src/Text/pages/TextPageObject.ts

```diff
@@ -22,10 +66,22 @@ class TextPageObject extends BasePage {
     return By(FIRST_TEXT_COMPONENT);
   }
 
-  get _secondaryComponent() {
```

Comment:
It's seems a bit strange to have two components return the same element (`_primaryComponent` and `_firstComponent`).

Rather than forcing a consumer of BasePage to use `_primaryComponent`, maybe we can make an overload that takes a different component as well.

If you look at `scrollToTestElement(component?: WebdriverIO.Element)` in basepage, we give the user the option to pass in a different component rather than using _primaryComponent.

We can also change `waitForPrimaryElementDisplayed()` to take in the same parameter and perform the same logic.

Then, in Text.spec.win.ts, instead of calling ` await TextPageObject.scrollToTestElement();` and `await TextPageObject.waitForPrimaryElementDisplayed(PAGE_TIMEOUT);`, we can call `await TextPageObject.scrollToTestElement(TextComponentSelector.First);` and `await TextPageObject.waitForPrimaryElementDisplayed(TextComponentSelector.First, PAGE_TIMEOUT);`

Then we can remove _primaryComponent completely from this file (or, if there are other functions like above, we can change them in BasePage as well)

Reply:
Or, on second thought, we'd want to scroll to the component lowest on the test page, so probably TextComponentSelector.Fourth

Reply:
>It's seems a bit strange to have two components return the same element (_primaryComponent and _firstComponent).

I agree. I don't think the _primary/secondaryComponent pattern works for everyone. While it provides a good example on how to manage more than one test element, I don't think we should make defining a _primary/secondaryComponent necessary. But requiring an implementation for _testPage/_pageButton/_pageName makes sense.

I created waitForElementDisplayed() in BasePage.ts to allow the user to specify a UI element instead and made `waitForPrimaryElementDisplayed()` call it with _primaryComponent so current users of `waitForPrimaryElementDisplayed` go unaffected.